src/WebJobs.Script/Description/DotNet/DotNetFunctionDescriptorProvider.cs

```diff
@@ -154,6 +155,12 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
                     descriptors.Add(new ParameterDescriptor(ScriptConstants.SystemExecutionContextParameterName, typeof(ExecutionContext)));
                 }
 
+                if (!descriptors.Any(p => p.Type == typeof(CancellationToken)))
+                {
+                    // Add the CancellationToken. We need this to ensure we can cancel even if the Function hasn't requested it.
```

Comment:
I may be being dense, but I still don't understand. Why do you need to add a cancellation token to the function? If the user hasn't defined their own then their function isn't coded to support cancellation, so why should we add one? (it won't be used anywhere).